eth-contracts/package.json

```diff
@@ -4,10 +4,10 @@
   "description": "CKB SPV proofs in Solidity",
   "scripts": {
     "compile": "hardhat compile",
-    "start-geth": "docker run -d --name geth-node --mount source=geth-dag,target=/root/.ethash -p 8545:8545 tockb/geth-priv:v1.9.23",
+    "start-geth": "docker run -d --name geth-node --mount source=geth-dag,target=/root/.ethash -p 8545:8545 tockb/geth-priv:v1.9.23 && ./wait-for-it.sh -t 0 127.0.0.1:8545 -- echo \"node is up\" && node scripts/waitForMiner.js",
     "stop-geth": "docker rm -f geth-node",
-    "pretest": "npm run start-geth && ./wait-for-it.sh -t 0 127.0.0.1:8545 -- echo \"node is up\" && node scripts/waitForMiner.js",
-    "test": "hardhat test --network geth",
+    "pretest": "npm run start-geth",
+    "test": "hardhat test --network geth || (npm run stop-geth && false)",
```

Comment:
Why?

```
|| (npm run stop-geth && false)
```

The posttest should be triggered automatically. And it should be ok even if the previous steps fails.

Reply:
The posttest won't be executed if test failed according to npm, what the script do is to make posttest executed even if test failed.

When we yarn test locally, we always need execute posttest whether the test succeed or not.